main/exercise/ptest_category.class.php

```diff
@@ -0,0 +1,476 @@
+                        description = '$description',
+                        color = '$color',
+                        position = $position
+                    WHERE id = $id AND c_id = ".$courseId;
+            Database::query($sql);
+
+            // item_property update
+            api_item_property_update(
+                $courseInfo,
+                TOOL_PTEST_CATEGORY,
+                $this->id,
+                'TestCategoryModified',
+                api_get_user_id()
+            );
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Gets the number of categories of exercise id=in_id.
+     *
+     * @param int $exerciseId
+     *
+     * @return int
+     */
+    public function getCategoriesExerciseNumber($exerciseId)
+    {
+        $table = Database::get_course_table(TABLE_QUIZ_CATEGORY_PTEST);
+        $exerciseId = (int) $exerciseId;
+        $sql = "SELECT count(*) AS nb
+                FROM $table
+                WHERE exercise_id = $exerciseId AND c_id=".api_get_course_int_id();
+        $res = Database::query($sql);
+        $row = Database::fetch_array($res);
+
+        return $row['nb'];
+    }
+
+    /**
+     * Return an array of all Category objects of exercise in the database
+     * If $field=="" Return an array of all category objects in the database
+     * Otherwise, return an array of all in_field value
+     * in the database (in_field = id or name or description).
+     *
+     * @param int    $exerciseId
+     * @param string $field
+     * @param int    $courseId
+     *
+     * @return array
+     */
+    public static function getCategoryListInfo($exerciseId, $field = '', $courseId = 0)
+    {
+        $exerciseId = (int) $exerciseId;
+        $courseId = empty($courseId) ? api_get_course_int_id() : (int) $courseId;
+
+        $table = Database::get_course_table(TABLE_QUIZ_CATEGORY_PTEST);
+        $categories = [];
+        if (empty($field)) {
+            $sql = "SELECT id FROM $table
+                    WHERE c_id = $courseId AND exercise_id = $exerciseId 
+                    ORDER BY position ASC";
+            $res = Database::query($sql);
+            while ($row = Database::fetch_array($res)) {
+                $category = new PTestCategory();
+                $categories[] = $category->getCategory($row['id'], $courseId);
+            }
+        } else {
+            $field = Database::escape_string($field);
+            $sql = "SELECT $field FROM $table
+                    WHERE c_id = $courseId AND exercise_id = $exerciseId 
+                    ORDER BY $field ASC";
+            $res = Database::query($sql);
+            while ($row = Database::fetch_array($res)) {
+                $categories[] = $row[$field];
+            }
+        }
+
+        return $categories;
+    }
+
+    /**
+     * @param FormValidator $form
+     * @param string        $action
+     */
+    public function getForm(&$form, $action = 'new')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $category_parent_list is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)